streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -153,4 +155,13 @@ public Cancellable schedule(final long interval, final PunctuationType type, fin
         return task.schedule(interval, type, callback);
     }
 
+    void setStreamTimeSupplier(final Supplier<Long> streamTimeSupplier) {
```

Comment:
rather than using `Supplier` could we use an interface that returns `long`? Avoid unnecessary boxing of `long` -> `Long`?

Reply:
Sure!